### DISCUSSION TITLE
CAKE 1.0.0 and VSIX project

### DISCUSSION THREAD

**Discussion by HStewartGitHome** [cake-build/cake#3187] [Q&A] [answered]
CAKE 1.0.0
Windows 10 64bit
Just Local currently

I notice Cake and wanted to try it on a visual studio extension project because it was so simple to use. I got some error so I thought maybe it was something specific to my own project and created a new project from Visual Studio and still got the errors when using using Cake. Note this that it built successfully within Visual Studio.

Just create using VSIX template (not empty) and try it out. Note - there is a known issue with VSIX build that was notice on stack overflow but can be solved by turning off the following option

Uncheck the “Deploy VSIX content to experimental” in the VSIX tab of your project properties. (this will allow you to build the project)

Here is output from my test

PS C:\Users\user\source\repos\VSIXProject1> dotnet cake

========================================
Clean
========================================
Restore
Determining projects to restore…
Restored C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj (in 260 ms).

========================================
Build
Microsoft ® Build Engine version 16.8.3+39993bd9d for .NET
Copyright © Microsoft Corporation. All rights reserved.

C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(1,17): error CS0234: The type or namespace name ‘VisualStudio’ does not exist in the namespace ‘Microsoft’ (are you missing an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(28,47): error CS0246: The type or namespace name ‘AsyncPackage’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(26,6): error CS0246: The type or namespace name ‘PackageRegistrationAttribute’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(26,6): error CS0246: The type or namespace name ‘PackageRegistration’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(26,26): error CS0246: The type or namespace name ‘UseManagedResourcesOnly’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(26,58): error CS0246: The type or namespace name ‘AllowsBackgroundLoading’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(44,102): error CS0246: The type or namespace name ‘ServiceProgressData’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]

Build FAILED.

C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(1,17): error CS0234: The type or namespace name ‘VisualStudio’ does not exist in the namespace ‘Microsoft’ (are you missing an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(28,47): error CS0246: The type or namespace name ‘AsyncPackage’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(26,6): error CS0246: The type or namespace name ‘PackageRegistrationAttribute’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(26,6): error CS0246: The type or namespace name ‘PackageRegistration’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(26,26): error CS0246: The type or namespace name ‘UseManagedResourcesOnly’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(26,58): error CS0246: The type or namespace name ‘AllowsBackgroundLoading’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1Package.cs(44,102): error CS0246: The type or namespace name ‘ServiceProgressData’ could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\source\repos\VSIXProject1\VSIXProject1\VSIXProject1.csproj]
0 Warning(s)
7 Error(s)

Time Elapsed 00:00:00.49
An error occurred when executing task ‘Build’.
Error: One or more errors occurred. (.NET Core CLI: Process returned an error (exit code 1).)
.NET Core CLI: Process returned an error (exit code 1).

**Comment by augustoproiete** [comment]
@HStewartGitHome Can you share the contents of your `build.cake` file? It's hard to help without knowing what are the commands you're executing...

Alternatively, does the output of any of these commands give you any clues?

```
dotnet cake --settings_showprocesscommandline=true
dotnet cake --verbosity=diagnostic
```

**Comment by HStewartGitHome** [comment]
Ok  I included what you ask.

I included them as files..  This is was my first attempt at Cake.  I am going try a normal (non VSIX) solution soon.

[build - Copy.txt](https://github.com/cake-build/cake/files/6031109/build.-.Copy.txt)

[cmds.txt](https://github.com/cake-build/cake/files/6031085/cmds.txt)

[diag.txt](https://github.com/cake-build/cake/files/6031111/diag.txt)

**Comment by nils-a** [comment, marked as answer]
@HStewartGitHome I get the exact same error, when I run `dotnet build` on a newly created VISX-Project. It seems that is not Cake specific.

However, building using `msbuild` works fine for me. So your build-step should probably utilize the `MSBuild` alias instead of the `DotNetCoreBuild` alias.

Something like:

```cakescript
 MSBuild(
    "VSIXProject1/VSIXProject1.sln",
     new MSBuildSettings 
     {
       Verbosity = Verbosity.Verbose,
       Configuration = configuration,
       PlatformTarget = PlatformTarget.MSIL
     });
```

should get you started.

**Reply by HStewartGitHome** [reply]
I try this and still have a problem, is possible some one to upload the files - maybe I have something wrong with syntax or system issue.

**Reply by nils-a** [reply]
like https://github.com/nils-a-demos/cake-vsix-sample ?

**Reply by HStewartGitHome** [reply]
Thanks I track down the issue to something to that I was using DotNetCoreRestore.   But VSIX are not .Net Core. I only play with Cake today and it Alias must be only for .Net Core projects.   Not sure when VSIX is going .Net Core, like next major version Visual Studio,  but alias  work nice for .Net Core projects.

Thanks.
